### DISCUSSION TITLE
How to move our player freely ?

### DISCUSSION THREAD

**Discussion by Deytooh** [naverz/zepeto-studio-kor#195] [5. Q&A]
QUESTION 1  -> 
Hi ~ 안녕하세요 ~ If english is not okay 한국어로도 쓸 수 있어서 더 편하면 말씀해주세요 ~ 
I am trying to make my character dash in the air, however, I find it was very hard to freely control the transform of my character.

Here is the code I tried. It works if I want to move him along x and z, the problem is that I have no control over y. when he is on the ground I cant modify y pos, and when he is in the air by Jump() method, I cant neither modify the y position... Help me please !

한국어 버전 : (이해못하면 ...죄송합니다 ㅋㅋㅋ )

캐릭터 대시 하게 만들고싶은데 y axis 도 움집여야돼는데 지금까지 해본 코드으로 못했어요 ! 방법이 있을까요 ?  
어떻게 자유로 캐릭테 움직일 수 있을까요 ? 

`

    
    public _isPlayerInitialized: bool = false;
    public _player: ZepetoPlayers["LocalPlayer"];
    public _playerTransform: Transform;

    Start() {

        ZepetoPlayers.instance.CreatePlayerWithZepetoId("...", "...", new SpawnInfo(), true);
        ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
            this._player = ZepetoPlayers.instance.LocalPlayer;
            this.transform.SetParent(this._player.zepetoPlayer.character.transform);
            this._playerTransform = this._player.zepetoPlayer.character.transform;
            this._isPlayerInitialized = true;
            
            //  this._player.zepetoPlayer.character.enabled = false;
        })
}`

`Update() {

        if (!this._isPlayerInitialized) return;

        if (Input.GetMouseButton(1)) {

            let vel = UtilsFunction.MultiplyVector(Vector3.up, Time.deltaTime);
            
            // this._player.zepetoPlayer.character.MoveContinuously(vel); //Doesnt work
            // this._player.zepetoPlayer.character.Move(vel);// Doesnt work
            //this._player.zepetoPlayer.character.characterController.Move(vel);// Doesnt work
            //this._player.zepetoPlayer.character.characterController.SimpleMove(vel); // Doesnt work
            //this._playerTransform.position = UtilsFunction.AddVector(this._playerTransform.position, vel); //Doenst work
            //this._playerTransform.Translate(vel); //Doenst work

        }
}`

QUESTION 2 ->
As you can see in my code, I made my own "UtilsFunction" class to code some functions I couldnt use with typescript like the simple 
`transform.position += Vector3.up` was not possible so I simply made AddVector method, I dont really like that, and I dont know typescript since I am focused only on the C# while developing on Unity. Is there some tips to make custom operators in typescript ? so I can be able to do transform.position += Vector ?

한국어 버전 : 
transform.position += Vector3.up 같은 코드 typescript에서 어떻게 할까요 ? " AddVector " 코드 썼는데 이렇게 하는거 별로 좋아하지않아서..

Thank you :) 좋은 하루 돼세요 ~

**Comment by x4cf** [comment]
ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.Move(Vector.Up);

**Reply by Deytooh** [reply]
Hi ~ 안녕하세요 ~ Thank you but it doesnt work :( 벌써 해봤지만 안돼요 이렇게 :(
